engineering_tools/carma

```diff
@@ -252,6 +252,70 @@ carma-config__status() {
     fi
 }
 
+carma__exec() {
+    local TARGET_IMAGE="usdotfhwastol/carma-platform:latest"
+    local COMMAND="bash"
+
+    # Check if the -i argument was provided
+    # If the image was specified then use that
+    if [ "$1" = "-i" ]; then 
+
+        if [ -z "$2" ]; then
+            echo " When -i is specified you must enter a carma docker image and tag."
+            exit -1
+        fi
+
+        local TARGET_IMAGE="$2"
+
+        shift
+        shift
+
+    # If the image was not specified use what was in the carma config
+    elif [ docker container inspect carma-config > /dev/null 2>&1 ]; then
+
+        local IMAGE_BACKUP=$TARGET_IMAGE
+        local SUPPORTED_ORANIZATIONS="usdotfhwastol\|usdotfhwastoldev"
+        local SUPPORTED_IMAGES="carma-platform:\|carma-messenger:"
+
+        # sed command finds the first instance of an image which is part of the set defined in SUPPORTED_ORGANIZATIONS and SUPPORTED_IMAGES.
+        # then returns the name of the image.  
+        local TARGET_IMAGE=$(docker run --rm --volumes-from carma-config:ro --entrypoint sh busybox:latest -c \
+            "sed -n -e '/image:\s*\($SUPPORTED_ORANIZATIONS\)\/\($SUPPORTED_IMAGES\)/ s/.*\image: *//p' docker-compose.yml | grep -m 1 '.*'")
+            
+
+        # If the sed resulted in an empty string then reset the tag
+        if [ -z "$TARGET_IMAGE" ]; then
+            local TARGET_IMAGE=$IMAGE_BACKUP
+        fi
+    
+    fi
+
+    # If a command was specified then assign it
+    if [ -n "$1" ]; then
+        local COMMAND="$@"
+    fi
+
+    # Docker setup here is based on Pierre Killy's docker-ros-box https://github.com/pierrekilly/docker-ros-box located here.
```

Comment:
Is this dependent again on the USERIDs lining up? Does the CARMA user in the container have to match the USERID for the current running X11 session?

Reply:
As far as I could tell its not. Worked both before and after I set the user id. But probably need more testing to be sure of the limitations.